Source/WebCore/dom/Document.cpp

```diff
@@ -9150,6 +9159,29 @@ void Document::canvasChanged(CanvasBase& canvasBase, const std::optional<FloatRe
         if (!changedRect)
             scheduleRenderingUpdate(RenderingUpdateStep::PrepareCanvasesForDisplay);
     }
+
+    if (!m_shouldDisableSleepForCanvasRendering || m_canvasChangedWhileDisablingSleep)
+        return;
+
+    if (m_sleepDisablerTimer.isActive()) {
+        m_canvasChangedWhileDisablingSleep = true;
+        return;
+    }
+
+    m_sleepDisablerTimer.startRepeating(5_s);
+    m_sleepDisabler = makeUnique<SleepDisabler>("com.apple.WebCore: Document playing audio with active canvas"_s, PAL::SleepDisabler::Type::Display);
+    m_canvasChangedWhileDisablingSleep = false;
```

Comment:
Nit: I think this would be slightly clearer if you set `m_canvasChangedWhileDisablingSleep = true` before checking to see if the timer is active, and don't set `m_canvasChangedWhileDisablingSleep = false` at all in this method.